### ISSUE TITLE
Issue with the new core/gallery custom attribute are getting undefined after activating new version of gutenberg.

### ISSUE THREAD

**Issue by AbhijitNage123** [WordPress/gutenberg#37675]
### Description

Hello there,
It seems that while testing the newest version of Gutenberg v1.12.0.
I found that the custom attributes are getting undefined for the new core/gallery.
I know that Gutenberg refactored the core/gallery with images block. I'm adding some attributes to the core gallery to customize it.
But I'm not getting the values of the old attributes now. Is there any way to get a previously-stored attribute value?

### Step-by-step reproduction instructions

Some screenshots - 
Before activating v1.12.0 - https://share.getcloudapp.com/bLuxrxXw
After activating v1.12.0 - https://share.getcloudapp.com/DOumgm5z

### Screenshots, screen recording, code snippet

```
/**
 * External Dependencies
 */
 import classnames from 'classnames';

 /**
  * WordPress Dependencies
  */
 const { __ } = wp.i18n;
 const { addFilter } = wp.hooks;
 const { Fragment }	= wp.element;
 const { InspectorAdvancedControls }	= wp.editor;
 const { createHigherOrderComponent } = wp.compose;
 const { ToggleControl, RangeControl } = wp.components;
 
 //restrict to specific block names
 const allowedBlocks = [ 'core/gallery' ];
 
 /**
  * Add custom attribute for mobile visibility.
  *
  * @param {Object} settings Settings for the block.
  *
  * @return {Object} settings Modified settings.
  */
 function addAttributes( settings ) {
     
     //check if object exists for old Gutenberg version compatibility
     //add allowedBlocks restriction
     if( typeof settings.attributes !== 'undefined' && allowedBlocks.includes( settings.name ) ){
     
         settings.attributes = Object.assign( settings.attributes, {
            gap: {
                type: 'number',
            },
            visibleOnMobile: { 
				type: 'boolean',
				default: true,
			}
         });
     
     }
 
     return settings;
 }
 
 /**
  * Add mobile visibility controls on Advanced Block Panel.
  *
  * @param {function} BlockEdit Block edit component.
  *
  * @return {function} BlockEdit Modified block edit component.
  */
 const withAdvancedControls = createHigherOrderComponent( ( BlockEdit ) => {
     return ( props ) => {
 
         const {
             name,
             attributes,
             setAttributes,
             isSelected,
         } = props;
 
         const {
            visibleOnMobile,
            gap
         } = attributes;
         console.log(attributes)
         
         return (
             <Fragment>
                 <BlockEdit {...props} />
                 { isSelected && allowedBlocks.includes( name ) &&
                     <InspectorAdvancedControls>
                         <ToggleControl
                             label={ __( 'Mobile Devices Visibity' ) }
                             checked={ !! visibleOnMobile }
                             onChange={ () => setAttributes( {  visibleOnMobile: ! visibleOnMobile } ) }
                             help={ !! visibleOnMobile ? __( 'Showing on mobile devices.' ) : __( 'Hidden on mobile devices.' ) }
                         />
                        <RangeControl
                            label={ __( 'Gap' ) }
                            value={ gap }
                            onChange={ (value) => setAttributes( {  gap: value } ) }
                            min={ 0 }
                            max={ 100 }
                            allowReset
                        />
                     </InspectorAdvancedControls>
                 }
 
             </Fragment>
         );
     };
 }, 'withAdvancedControls');
 
 /**
  * Add custom element class in save element.
  *
  * @param {Object} extraProps     Block element.
  * @param {Object} blockType      Blocks object.
  * @param {Object} attributes     Blocks attributes.
  *
  * @return {Object} extraProps Modified block element.
  */
 function applyExtraClass( extraProps, blockType, attributes ) {
 
     const { visibleOnMobile } = attributes;
     
     //check if attribute exists for old Gutenberg version compatibility
     //add class only when visibleOnMobile = false
     //add allowedBlocks restriction
     if ( typeof visibleOnMobile !== 'undefined' && !visibleOnMobile && allowedBlocks.includes( blockType.name ) ) {
         extraProps.className = classnames( extraProps.className, 'mobile-hidden' );
     }
 
     return extraProps;
 }
 
 //add filters
 
 addFilter(
     'blocks.registerBlockType',
     'editorskit/custom-attributes',
     addAttributes
 );
 
 addFilter(
     'editor.BlockEdit',
     'editorskit/custom-advanced-control',
     withAdvancedControls
 );
 
 addFilter(
     'blocks.getSaveContent.extraProps',
     'editorskit/applyExtraClass',
     applyExtraClass
 );```

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
@glendaviesnz curious if you have any insights here!

**Comment by glendaviesnz** [comment]
Thanks for reporting this @AbhijitNage123. We didn't account for custom attributes in the Gallery block migrations. I have a potential fix for this [here](https://github.com/WordPress/gutenberg/pull/37812). 

It was good to pick this up while we still have a chance to get a fix for this into 5.9!
